test_scripts/API/VehicleData/GearStatus/common.lua

```diff
@@ -0,0 +1,441 @@
+local events = require('events')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.defaultProtocolVersion = 2
+
+--[[ Local Variables ]]
+local m = {}
+local hashId = {}
+
+--[[ Shared Functions ]]
+m.Title = runner.Title
+m.Step = runner.Step
+m.getPreloadedPT = actions.sdl.getPreloadedPT
+m.setPreloadedPT = actions.sdl.setPreloadedPT
+m.registerApp = actions.registerApp
+m.registerAppWOPTU = actions.registerAppWOPTU
+m.activateApp = actions.activateApp
+m.getMobileSession = actions.getMobileSession
+m.getHMIConnection = actions.getHMIConnection
+m.backupPreloadedPT = actions.sdl.backupPreloadedPT
+m.restorePreloadedPT = actions.sdl.restorePreloadedPT
+m.cloneTable = utils.cloneTable
+m.getConfigAppParams = actions.getConfigAppParams
+m.start = actions.start
+m.policyTableUpdate = actions.policyTableUpdate
+m.getAppsCount = actions.getAppsCount
+m.getParams = actions.app.getParams
+m.deleteSession = actions.mobile.deleteSession
+m.connectMobile = actions.mobile.connect
+
+m.gearStatusData = {
+  userSelectedGear = "NINTH",
+  actualGear = "TENTH",
+  transmissionType = "MANUAL"
+}
+
+m.subUnsubResponse = {
+  dataType = "VEHICLEDATA_GEARSTATUS",
+  resultCode = "SUCCESS"
+}
+
+m.prndlData = "PARK"
+
+--[[ Common Functions ]]
+
+--[[ @updatePreloadedPT: Update preloaded file with additional permissions for GearStatus
+--! @parameters:
+--! pGroups: table with additional updates (optional)
+--! @return: none
+--]]
+function m.updatePreloadedPT(pGroups)
+  local pt = m.getPreloadedPT()
+  if not pGroups then
+    pGroups = {
+      rpcs = {
+        GetVehicleData = {
+          hmi_levels = { "BACKGROUND", "LIMITED", "FULL" },
+          parameters = { "gearStatus", "prndl" }
+        },
+        OnVehicleData = {
+          hmi_levels = { "BACKGROUND", "LIMITED", "FULL" },
+          parameters = { "gearStatus", "prndl" }
+        },
+        SubscribeVehicleData = {
+          hmi_levels = { "BACKGROUND", "LIMITED", "FULL" },
+          parameters = { "gearStatus", "prndl" }
+        },
+        UnsubscribeVehicleData = {
+          hmi_levels = { "BACKGROUND", "LIMITED", "FULL" },
+          parameters = { "gearStatus", "prndl" }
+        }
+      }
+    }
+  end
+  pt.policy_table.functional_groupings["NewTestCaseGroup"] = pGroups
+  pt.policy_table.app_policies["default"].groups = { "Base-4", "NewTestCaseGroup" }
+  pt.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  m.setPreloadedPT(pt)
+end
+
+--[[ @preconditions: Clean environment and optional backup and update of sdl_preloaded_pt.json file
+--! @parameters:
+--! isPreloadedUpdate: if true then sdl_preloaded_pt.json file will be updated, otherwise - false
+--! @return: none
+--]]
+function m.preconditions(isPreloadedUpdate)
+  if isPreloadedUpdate == nil then isPreloadedUpdate = true end
+  actions.preconditions()
+  if isPreloadedUpdate == true then
+    m.backupPreloadedPT()
```

Comment:
Please remove `m.backupPreloadedPT` because this function is called in m`.setPreloadedPT`